### ISSUE TITLE
`bun update --save` doesn't save to `package.json`

### ISSUE THREAD

**Issue by noClaps** [oven-sh/bun#4312]
### What version of Bun is running?

0.8.0+c60385716b7a7ac9f788cdf7dfe37250321e0670

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

My original dependencies:

```json
  "dependencies": {
    "@astrojs/cloudflare": "^6.8.0",
    "astro": "^2.10.12",
    "astro-icon": "^0.8.1",
    "nanostores": "^0.9.3",
    "sass": "^1.66.1"
  },
  "devDependencies": {
    "@types/node": "^20.5.1",
    "wrangler": "^3.5.1"
  }
```

Run `bun update --save`. 

Run `bun pm ls`:

```
/Users/user/Programming/web node_modules (1041)
├── @astrojs/cloudflare@6.8.1
├── @types/node@20.5.6
├── astro@2.10.14
├── astro-icon@0.8.1
├── nanostores@0.9.3
├── sass@1.66.1
└── wrangler@3.6.0
```

`package.json` after update:

```json
  "dependencies": {
    "@astrojs/cloudflare": "^6.8.0",
    "astro": "^2.10.12",
    "astro-icon": "^0.8.1",
    "nanostores": "^0.9.3",
    "sass": "^1.66.1"
  },
  "devDependencies": {
    "@types/node": "^20.5.1",
    "wrangler": "^3.5.1"
  }
```

### What is the expected behavior?

Running `bun update --save` should update the package.json file.

Expected `package.json` after update:

```json
  "dependencies": {
    "@astrojs/cloudflare": "^6.8.1",
    "astro": "^2.10.14",
    "astro-icon": "^0.8.1",
    "nanostores": "^0.9.3",
    "sass": "^1.66.1"
  },
  "devDependencies": {
    "@types/node": "^20.5.6",
    "wrangler": "^3.6.0"
  }
```

### What do you see instead?

Actual `package.json` after `bun update --save`:

```json
  "dependencies": {
    "@astrojs/cloudflare": "^6.8.0",
    "astro": "^2.10.12",
    "astro-icon": "^0.8.1",
    "nanostores": "^0.9.3",
    "sass": "^1.66.1"
  },
  "devDependencies": {
    "@types/node": "^20.5.1",
    "wrangler": "^3.5.1"
  }
```

**Comment by riywo** [comment]
A workaround would be using `yarn`:

```
yarn upgrade --latest && bun install --yarn
```

But I would not like to use `yarn` as it's tooooooo slow! (or `bun` is toooo fast!)

**Comment by noClaps** [comment]
Issue persists in Bun 1.0

**Comment by silvenon** [comment]
This is unfortunate because the `bun update` command description claims that it will update `package.json`:

> Update outdated dependencies & save to package.json

And `bun add <pkg>@latest` changes the version to literally `latest`, rather than the last version, so there's no workaround other than manually updating `package.json`.

**Comment by noClaps** [comment]
There's the `npm-check-updates` package that could be a workaround, but it would be nice to have Bun handle updates properly.

**Comment by ctison** [comment]
As @noClaps mentionnend, [npm-check-updates](https://www.npmjs.com/package/npm-check-updates) is a nice tool to update `package.json` dependencies:
```sh
bun x npm-check-updates -ui
```
It even works in workspaces!
```sh
bun x npm-check-updates -ui -ws --root
```

**Edit**: Added the `--root` flag to update the root `package.json` as well.

**Comment by ctjlewis** [comment]
Commenting just to follow. Using `npm-check-updates` and updating manually in meantime.

**Comment by sindelio** [comment]
Following this task as well, +1 to prioritize it

**Comment by silvenon** [comment]
Actually `npm-check-updates` is working just fine for me in the meantime:

```sh
bunx npm-check-updates -ui # short form
bunx npm-check-updates --upgrade --interactive # long form
```

I was weary of solutions like these, but no need, whatever works, and this lets me continue enjoying Bun ❤️

**Comment by MyWay** [comment]
> Actually `npm-check-updates` is working just fine for me in the meantime:
> 
> ```shell
> bunx npm-check-updates -ui # short form
> bunx npm-check-updates --upgrade --interactive # long form
> ```
> 
> I was weary of solutions like these, but no need, whatever works, and this lets me continue enjoying Bun ❤️

Doing so you'd have to re-run bun install to update bun.lockb file, though?

**Comment by silvenon** [comment]
I don't think so, `npm-check-updates` supports Bun.

**Comment by touhidurrr** [comment]
As of `1.1.4`, I am facing problems with:
1. `bun update`
2. `bun update --latest`
3. `bun update -f`
4. `bun update -f --latest`

None of these commands update `package.json`. Not to mention, adding `--save` flag does not help. But this should not be required.

**Comment by silvenon** [comment]
@touhidurrr yes, that has been established 😄 there's nothing in the v1.1.4 release that claims to be fixing this.

**Comment by touhidurrr** [comment]
> @touhidurrr yes, that has been established 😄 there's nothing in the v1.1.4 release that claims to be fixing this.

ik. i just provided an update on the current status of the issue. Also, letting everyone know that no version or flag of this command fixes the problem, a point that i feel has not been discussed adequately.

**Comment by dylan-conway** [comment]
Fixed in #11340. With this change, `bun update` will update versions in package.json by default. You can use `--no-save` for the original behavior.
